include/rcutils/thread_attr.h

```diff
@@ -0,0 +1,146 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCUTILS__THREAD_ATTR_H_
+#define RCUTILS__THREAD_ATTR_H_
+
+#include "rcutils/visibility_control.h"
+
+#include "rcutils/allocator.h"
+#include "rcutils/macros.h"
+#include "rcutils/types/rcutils_ret.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+typedef enum rcutils_thread_scheduling_policy_e
+{
+  RCUTILS_THREAD_SCHEDULING_POLICY_UNKNOWN  = 0,
+  RCUTILS_THREAD_SCHEDULING_POLICY_FIFO     = 1,
+  RCUTILS_THREAD_SCHEDULING_POLICY_RR       = 2,
+  RCUTILS_THREAD_SCHEDULING_POLICY_SPORADIC = 3,
+  RCUTILS_THREAD_SCHEDULING_POLICY_OTHER    = 4,
+  RCUTILS_THREAD_SCHEDULING_POLICY_IDLE     = 5,
+  RCUTILS_THREAD_SCHEDULING_POLICY_BATCH    = 6,
+  RCUTILS_THREAD_SCHEDULING_POLICY_DEADLINE = 7
+} rcutils_thread_scheduling_policy_t;
+
+typedef struct rcutils_thread_attr_s
+{
+  /// Thread core affinity
+  int core_affinity;
+  /// Thread scheduling policy.
+  rcutils_thread_scheduling_policy_t scheduling_policy;
+  /// Thread priority.
+  int priority;
+  /// Thread name
+  char const * name;
+} rcutils_thread_attr_t;
+
+/// Hold thread attribute rules.
+typedef struct rcutils_thread_attrs_s
+{
+  /// Private implementation array.
+  rcutils_thread_attr_t * attributes;
+  /// Number of threads attribute
+  size_t num_attributes;
+  /// Number of threads attribute capacity
+  size_t capacity_attributes;
+  /// Allocator used to allocate objects in this struct
+  rcutils_allocator_t allocator;
+} rcutils_thread_attrs_t;
+
+/**
+ * \brief Return a rcutils_thread_attrs_t struct with members initialized to zero value.
+ * \return a rcutils_thread_attrs_t struct with members initialized to zero value.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_thread_attrs_t
+rcutils_get_zero_initialized_thread_attrs(void);
+
+/**
+ * \brief Initialize list of thread attributes.
+ * \param[out] thread_attrs list of thread attributes to be initialized
+ * \param[in] allocator memory allocator to be used
+ * \return #RCUTILS_RET_OK if the structure was initialized succeessfully, or
+ * \return #RCUTILS_RET_INVALID_ARGUMENT if any function arguments are invalid, or
+ * \return #RCUTILS_RET_BAD_ALLOC if allocating memory failed, or
+ * \return #RCUTILS_RET_ERROR an unspecified error occur.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_ret_t
+rcutils_thread_attrs_init(
+  rcutils_thread_attrs_t * thread_attrs,
+  rcutils_allocator_t allocator);
+
+/**
+ * \brief Initialize list of thread attributes with a capacity.
+ * \param[out] thread_attrs list of thread attributes to be initialized
+ * \param[in] allocator memory allocator to be used
```

Comment:
```suggestion
 * \param[in] allocator memory allocator to be used
 * \param[in] capacity ...
```

Reply:
Thanks for your suggestion.
I've addressed it in https://github.com/ros2/rcutils/pull/424/commits/c73ab591d9ed44247813ea56e5f76c0601b09a08.